src/m365/viva/commands/engage/engage-role-member-list.spec.ts

```diff
@@ -0,0 +1,209 @@
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.ENGAGE_ROLE_MEMBER_LIST);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('defines correct properties for the default output', () => {
+    assert.deepStrictEqual(command.defaultProperties(), ['id', 'userId']);
+  });
+
+  it('fails validation if roleId is not a valid GUID', () => {
+    const actual = commandOptionsSchema.safeParse({
+      roleId: 'invalid'
+    });
+    assert.notStrictEqual(actual.success, true);
+  });
+
+  it('passes validation if roleId is a valid GUID', () => {
+    const actual = commandOptionsSchema.safeParse({
+      roleId: roleId
+    });
+    assert.strictEqual(actual.success, true);
+  });
+
+  it('passes validation if roleName is specified', () => {
+    const actual = commandOptionsSchema.safeParse({
+      roleName: roleName
+    });
+    assert.strictEqual(actual.success, true);
+  });
+
+  it('fails validation if both roleId and roleName are specified', () => {
+    const actual = commandOptionsSchema.safeParse({
+      roleId: roleId,
+      roleName: roleName
+    });
+    assert.notStrictEqual(actual.success, true);
+  });
+
+  it('fails validation if neither roleId nor roleName is specified', () => {
+    const actual = commandOptionsSchema.safeParse({});
+    assert.notStrictEqual(actual.success, true);
+  });
+
+  it(`should get a list of users assigned to a Viva Engage role specified by id`, async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/beta/employeeExperience/roles/${roleId}/members`) {
+        return {
+          "value": [
+            {
+              "id": "1f5595b2-aa07-445d-9801-a45ea18160b2",
+              "createdDateTime": "2024-05-22T15:43:08.368Z",
+              "userId": "1f5595b2-aa07-445d-9801-a45ea18160b2"
+            },
+            {
+              "id": "717f1683-00fa-488c-b68d-5d0051f6bcfa",
+              "createdDateTime": "2025-07-16T02:51:22.602Z",
+              "userId": "717f1683-00fa-488c-b68d-5d0051f6bcfa"
+            }
+          ]
+        };
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: { verbose: true, roleId: roleId }
+    });
+
+    assert(
+      loggerLogSpy.calledOnceWith([
+        {
+          "id": "1f5595b2-aa07-445d-9801-a45ea18160b2",
+          "createdDateTime": "2024-05-22T15:43:08.368Z",
+          "userId": "1f5595b2-aa07-445d-9801-a45ea18160b2"
+        },
+        {
+          "id": "717f1683-00fa-488c-b68d-5d0051f6bcfa",
+          "createdDateTime": "2025-07-16T02:51:22.602Z",
+          "userId": "717f1683-00fa-488c-b68d-5d0051f6bcfa"
+        }
+      ])
+    );
+  });
+
+  it(`should get a list of users assigned to a Viva Engage role specified by names`, async () => {
+    sinon.stub(vivaEngageRole, 'getRoleIdByName').resolves(roleId);
```

Comment:
Let's use something like `.withParameters` to ensure the right parameter is passed.